coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -497,13 +487,14 @@ private RexNode adjustReturnTypeWithCast(RexBuilder rexBuilder, RexNode call) {
       }
       final String lowercaseOperatorName = ((RexCall) call).getOperator().getName().toLowerCase(Locale.ROOT);
       final ImmutableMap<String, RelDataType> operatorsToAdjust =
-          ImmutableMap.of("date_diff", typeFactory.createSqlType(INTEGER), "cardinality",
+          ImmutableMap.of("datediff", typeFactory.createSqlType(INTEGER), "cardinality",
               typeFactory.createSqlType(INTEGER), "ceil", typeFactory.createSqlType(BIGINT), "ceiling",
               typeFactory.createSqlType(BIGINT), "floor", typeFactory.createSqlType(BIGINT));
       if (operatorsToAdjust.containsKey(lowercaseOperatorName)) {
         return rexBuilder.makeCast(operatorsToAdjust.get(lowercaseOperatorName), call);
       }
-      if (configs.getOrDefault(CAST_DATEADD_TO_STRING, false) && lowercaseOperatorName.equals("date_add")) {
+      if ((configs.getOrDefault(CAST_DATEADD_TO_STRING, false) && lowercaseOperatorName.equals("date_add"))
+          || (configs.getOrDefault(CAST_DATESUB_TO_STRING, false) && lowercaseOperatorName.equals("date_sub"))) {
```

Comment:
Can we also move `adjustReturnTypeWithCast` from RelNode layer to SqlNode layer (`TrinoSqlUDFConverter`)? I think we don't need to introduce `CAST_DATESUB_TO_STRING` config if we do that.

Reply:
After checking the code again, it looks like we don't need to add a new configuration parameter of `CAST_DATESUB_TO_STRING` without moving this logic into SqlNode layer now, as `date_sub` function is converted into `date_add` function anyway.